learningequality/studio
# Change event log

## Summary
### Description of the change(s) you made
* Implements a change event log on the server side
* Makes all change application run inside an async task on the server side
* Separates out user specific and channel specific changes
* Allows syncing back of changes relevant to channels and users


### Manual verification steps performed
* Do edits
* Save edits
* Confirm they get synced
* Confirm they get applied
* Confirm they get removed from the frontend

## Reviewer guidance
### How can a reviewer test these changes?
Please test the basic flow for now.

Other thoughts:
* We could now sync task data through the repeatedly polling sync endpoint which might save us a request or two - sound worth it?
* There is some follow up that would be useful here for storing copying task info on the contentnode object to more robustly track the association
* What documentation updates would be helpful here, what is unclear in the code and could be better commented?

### Are there any risky areas that deserve extra testing?
EVERYTHING!


## References
Fixes #3034 
Fixes #2828 


## Comments


----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
